examples/mi-mctp.c

```diff
@@ -162,9 +164,66 @@ static int do_controllers(nvme_mi_ep_t ep)
 	return 0;
 }
 
+static const char *__copy_id_str(const void *field, size_t size,
+				 char *buf, size_t buf_size)
+{
+	assert(size < buf_size);
+	strncpy(buf, field, size);
+	buf[size] = '\0';
+	return buf;
+}
+
+#define copy_id_str(f,b) __copy_id_str(f, sizeof(f), b, sizeof(b))
+
+int do_identify(nvme_mi_ep_t ep, int argc, char **argv)
+{
+	struct nvme_mi_ctrl *ctrl;
+	struct nvme_id_ctrl id;
+	uint16_t ctrl_id;
+	char buf[41];
+	int rc, tmp;
+
+	if (argc != 2) {
+		fprintf(stderr, "no controller ID specified\n");
+		return -1;
+	}
+
+	tmp = atoi(argv[1]);
+	if (tmp < 0 || tmp > 0xffff) {
+		fprintf(stderr, "invalid controller ID\n");
+		return -1;
+	}
+
+	ctrl_id = tmp & 0xffff;
+
+	ctrl = nvme_mi_init_ctrl(ep, tmp);
```

Comment:
possible memleak. ctrl is not free() at last

Reply:
This is now handled through `nvme_mi_close_ctrl`.